### PR TITLE
chore(recordings): analytics for playback errors

### PR DESCRIPTION
**Pull request by rcmarron**
## Problem

We don't have great analytics about how often the recording player crashes or has an excessive number of warnings (indicative of broken playback)

## Changes

* Remove the `source` stuff from the existing analytics events because it was complex and redundant with `pathName`
* Add a new event `reportRecordingViewedSummary` that has:
    * viewed_time_ms
    * recording_duration_ms
    * recording_age_days
    * meta_data_load_time_ms
    * first_snapshot_load_time_ms
    * first_snapshot_and_meta_load_time_ms
    * all_snapshots_load_time_ms
    * rrweb_warning_count
    * error_count
* To get the rrweb warning count, we monkey patch `console.warn`
* To get the error count, we add an error listener when the recording is open. Not a perfect metric but good enough. We also send these errors to sentry.
👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

Watched recordings and verified the analytics events perform as expected. Specifically tests that:
* Calling `console.warn('[replayer] mumbo jumbo') increments the warning count
* Causing an error in the console during play back increments the error count

### CODE REVIEW

File: frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.test.ts [hunk 1]
```diff
@@ -208,6 +207,7 @@ describe('sessionRecordingDataLogic', () => {
         it('server error mid-fetch', async () => {
             const firstNext = `${EVENTS_SESSION_RECORDING_EVENTS_ENDPOINT}?person_id=1&before=2021-10-28T17:45:12.128000Z&after=2021-10-28T16:45:05Z`
             silenceKeaLoadersErrors()
+            jest.spyOn(api, 'get')
```
Comment (rcmarron) [inline frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.test.ts:210]: Note: The changes in this file (outside of removing `RecordingWatchedSource`) clean up some test error logs that are un-related to this PR, but needed cleaning

File: frontend/src/scenes/session-recordings/player/metaLogic.test.ts [hunk 2]
```diff
@@ -64,7 +67,10 @@ describe('metaLogic', () => {
                 },
             })
             sessionRecordingsTableLogic().mount()
+
+            silenceKeaLoadersErrors()
```
Comment (rcmarron) [inline frontend/src/scenes/session-recordings/player/metaLogic.test.ts:71]: Note: Same here, the changes in this file also just clean up some test error logs that are un-related to this PR, but needed cleaning

File: frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.test.ts [hunk 3]
```diff
@@ -279,8 +280,8 @@ describe('sessionRecordingDataLogic', () => {
             ]
 
         it('no next url', async () => {
-            await expectLogic(logic, () => {
-                logic.actions.loadRecordingSnapshots()
+            await expectLogic(logic, async () => {
+                await logic.actions.loadRecordingSnapshots()
```
Comment (alexkim205) [inline frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.test.ts:284]: nit: Don't think we need an async here?

File: frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.test.ts [hunk 4]
```diff
@@ -318,20 +320,23 @@ describe('sessionRecordingDataLogic', () => {
                     }
                 })
 
-            await expectLogic(logic, () => {
-                logic.actions.loadRecordingSnapshots()
+            await expectLogic(logic, async () => {
+                await logic.actions.loadRecordingSnapshots()
             })
```
Comment (alexkim205) [inline frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.test.ts:325]: nit: Here as well

File: frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts [hunk 5]
```diff
@@ -487,10 +503,54 @@ export const sessionRecordingPlayerLogic = kea<sessionRecordingPlayerLogicType>(
...
         beforeUnmount: () => {
             values.player?.replayer?.pause()
             actions.setPlayer(null)
+            if (cache.originalWarning) {
+                console.warn = cache.originalWarning
+            }
+            if (cache.errorHandler) {
+                window.removeEventListener('error', cache.errorHandler)
+            }
+            eventUsageLogic.actions.reportRecordingViewedSummary({
```
Comment (alexkim205) [inline frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts:516]: Does it make sense to have the same reporting happening wherever we mount the sessionRecordingPlayer logic? I think whenever sessionRecordingPlayer is unmounted, the corresponding sessionRecordingDataLogic should be unmounted as well, which would make this listener redundant. 

If we remove this listener, we can also clean up boilerplate by removing `loadMetaTimeMs`, `loadFirstSnapshotTimeMs` and `loadAllSnapshotTimeMs` and calculating those just in time on sessionRecordingData unmount.

**Review by alexkim205 — CHANGES_REQUESTED** [review]
left some comments!

File: frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.test.ts (hunk 3 above)
Comment (rcmarron) [reply frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.test.ts:284]: Ah good catch - added these in my battle against the test errors 🙈

File: frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts [hunk 6]
```diff
@@ -167,14 +178,18 @@ export const sessionRecordingPlayerLogic = kea<sessionRecordingPlayerLogicType>(
...
+                const replayer = new Replayer(values.sessionPlayerData.snapshotsByWindowId[windowId], {
+                    root: values.rootFrame,
+                    triggerFocus: false,
+                    insertStyleRules: [
+                        `.ph-no-capture {   background-image: url("data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTYiIGhlaWdodD0iMTYiIHZpZXdCb3g9IjAgMCAxNiAxNiIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPHJlY3Qgd2lkdGg9IjE2IiBoZWlnaHQ9IjE2IiBmaWxsPSJibGFjayIvPgo8cGF0aCBkPSJNOCAwSDE2TDAgMTZWOEw4IDBaIiBmaWxsPSIjMkQyRDJEIi8+CjxwYXRoIGQ9Ik0xNiA4VjE2SDhMMTYgOFoiIGZpbGw9IiMyRDJEMkQiLz4KPC9zdmc+Cg=="); }`,
+                    ],
+                })
+                actions.setPlayer({ replayer, windowId })
+            } catch (e) {
+                console.error('blahalhah', e)
```
Comment (alexkim205) [inline frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts:191]: oo found something

Comment (rcmarron) [reply frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts:191]: 🙈

**Review by alexkim205 — APPROVED** [review]

File: frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts (hunk 5 above)
Comment (rcmarron) [reply frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts:511]: Alex and I touched base on this one. We still want the analytics from `sessionRecordingDataLogic` to fire for now until we move graphs to the new stuff. So having some similar logic in both places makes sense for now.

**Pull request merged**